soroban-auth/tests/test.rs

```diff
@@ -30,6 +30,7 @@ fn verify_and_consume_nonce(e: &Env, id: &Identifier, expected_nonce: &BigInt) {
             if BigInt::zero(&e) != expected_nonce {
                 panic!("nonce should be zero for Contract")
             }
+            return;
         }
         _ => {}
     }
```

Comment:
Because this match is an early exit for a single case, I think it's more common to be express as an if let, or to include the rest of the function inside the else case.

Reply:
I couldn't figure out a good way to get the enum discriminant, which is why I went with the match. https://doc.rust-lang.org/std/mem/fn.discriminant.html is available, but it's in the std crate.

Reply:
I suggest using `matches!`:
```rust
    if matches!(id, Identifier::Contract(_)) {
        if BigInt::zero(&e) != expected_nonce {
            panic!("nonce should be zero for Contract")
        }
        return;
    }
```

Reply:
That's what I needed. Updated. Thanks!